.github/workflows/test-external-links-cron-job.yml

```diff
@@ -0,0 +1,26 @@
+name: Testing External Links Cron Job
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 16 * * *"  # Daily at noon EST
+
+jobs:
+  build-and-test:
+    name: Testing External Links
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - run: git fetch --prune --unshallow --tags
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Install package for API docs
+        run: pip install .[docs]
+      - name: Test External Links
+      # This command runs Sphinx's linkcheck with a configuration override to ensure no links are ignored during the check.
+      # The -D option here explicitly sets linkcheck_ignore to an empty list, overriding any ignore patterns set in conf.py.
+        run: sphinx-build -b linkcheck -D linkcheck_ignore=[] ./docs ./docs/_build/
```

Comment:
Wait, actually, could you add a final step here to do the email notification? And give it a email header about neuroconv external links?

Reply:
I have no idea how to do that.

Reply:
I actually would like to not receive emails myself.

Reply:
I want the action to be useful for avoiding adding new broken links. That function is less likely to be performed if the action is failing all the time as it is now.  

At the same time, you want to keep the action that is failing all the time to see if at some point it will stop failing. 

I am having two actions for these two different purposes.

Reply:
> I want the action to be useful for avoiding adding new broken links.

PR reviewers should also ensure links work as expected, so it's not like the CI should exclusively be relied on for that. It's just a nicety when it works

> I am having two actions for these two different purposes.

Except the burden of maintaining the 'ignore' list to achieve that 'passing' status is yet another thing I need to deal with now, instead of just resolving the source problematic links. I'd only be willing to add this layer if I can get help maintaining it

Reply:
> PR reviewers should also ensure links work as expected, so it's not like the CI should exclusively be relied on for that. It's just a nicety when it works

If we can make a better classifier to help us I don't see why we shouldn't. I classifier with less false positives is better.

>Except the burden of maintaining the 'ignore' list to achieve that 'passing' status is yet another thing I need to deal with now, instead of just resolving the source problematic links. I'd only be willing to add this layer if I can get help maintaining it

I am fine adding links that start failing to the list myself.

>  That happens a lot more often than you think lol

Could be. I don't think the stackoverflow ones can or will be solved. But in any case, having a link that is working on the list does not have any downside.

Reply:
But do we agree that this notification will be added to the action that runs in the PRs right?  The one that uses the ignore list.

If so, that makes sense to me and I will add it.

Reply:
The other one, the daily (or coarser CRON) that checks current links on `main`. We would not want notifications from pushes to PRs lol

Reply:
No, then we disagree. I think that that signal is not useful. And is not related to what I volunteer for: adding links to the list that ar failing.

I think getting a notification that the links are failing everyday is not a useful signal. Why do you want that?

Reply:
The entire origin of these CI checks is to ensure that links on `main` (in our public docs) do not go out of date or experience other disruptions (and yes, generally, SSL certificate issues would fall under that category, since it could indicate any number of problems or reasons why we wouldn't want to send users to that website through our own)

Links are a live service; their health must be checked periodically.

A PR-based workflow helps inform users that a link being added to a new PR needs attention; no one needs to be notified about that besides the people writing and reviewing the PR in question.

A daily-based workflow ensure the links stays good after merging, that's the maintenance cost to that

Your suggestion to make the PR-based workflow more effective by adding a specific ignore list adds a cost of maintenance (for items like the Buzsaki link; the alternative on stackoverflow is also to just remove the link entirely to make it a non-issue) since it should be periodically checked to see if any can be taken off of ignore, or likewise on regular daily failures if something needs to be either fixed or added to ignore if no other options. That's what I'm saying I would want help with in that case

Reply:
To satisfy your critieria here:

> A daily-based workflow ensure the links stays good after merging, that's the maintenance cost to that

I see two options to move forward:
* We make the action with the ignore list also a cron job and we add the notification to that.
* We just remove the Buzsaki links and the stackoverflow ones and we leave the current action as it is.

The things that I don't find value on having are:
1) An action that is always failing in the PR. Makes the work of non-adding broken links harder.
2) A daily action that is always failing and give us notification on that.

I though that you were interested on having the action without ignore list because you want to check if at some point the links start working but maybe I am wrong.

What do you think?

Reply:
OK, I prefer number 1. 

I think it will be more flexible long term. 

Buzsaki might be fixed, I don't think Stackoverflow will be fixed as I don't think they want bots accessing their links after the AI developments of the last year.

Do we have something now that is doing this?
> A daily-based workflow ensure the links stays good after merging, that's the maintenance cost to that

Reply:
Ah, OK, the current configuration is exactly like:

https://github.com/catalystneuro/neuroconv/blob/a4f9b7581eb58301f7f868dc1c101b454e0c878f/.github/workflows/dailies.yml#L1-L37

That is, the dallies will call the current check for the links and test it as I proposed above. I also added the codecov token in the dallies which is causing a failure in the CI as of  now.